test/port-finder-socket-test.js

```diff
@@ -69,7 +69,15 @@ function stopServers(callback, index) {
 }
 
 function cleanup(callback) {
-  fs.rmdirSync(badDir);
+  if (fs.existsSync(path.join(badDir, 'deeply', 'nested'))) {
+    fs.rmdirSync(path.join(badDir, 'deeply', 'nested'));
+  }
+  if (fs.existsSync(path.join(badDir, 'deeply'))) {
+    fs.rmdirSync(path.join(badDir, 'deeply'));
+  }
+  if (fs.existsSync(badDir)) {
+    fs.rmdirSync(badDir);
+  }
```

Comment:
Without adding the `if (fs.existsSync(...))` checks, when my new test failed on `modernize`, the `fs.rmdirSync` call would throw as the nested directory doesn't exist, and so then `stopServers` wouldn't be called and then vows would hang as there'd be sockets that were never cleaned up.

Reply:
Unfortunately, that function was added in Node 5, so until 0.12 - 4.x are no longer tested, I couldn't incorporate that. While I do think could probably revisit it later once that does happen, good to expand the test suite to help ensure no breaking changes on modernizing (well, besides dropping the old versions).

Reply:
@MasterOdin - so the issue where the left over bad folders and sockets inside fixtures occurred again - only once, this time on a cold mac m1. Recall though, last night it happened (consistently) inside docker for mac, intel x86

I'm noting it, I'll revisit, I may know why. 

Thoughts welcome. Forgive the short note, snooze time.

Reply:
it also passed completely for me earlier. 

maybe i botched it, i'll check tomorrow. 

feel free to check if ur curiuos

Reply:
@MasterOdin 

unfortunately the recursive deletion code didn't work - i experienced a hang and left over dirty file system in the test directory 

what u think about using  [fs.mkdtemp(prefix[, options], callback)](https://nodejs.org/api/fs.html#fsmkdtempprefix-options-callback) as a lifeline here? If people are using node < 5, they better have their own npm server :)  

its the best option for the tradeoff curve

(ps - to reproduce what i see, try in docker, rebuild each time you run, you'll notice - its not consistent below node 12 ish)